play-json/shared/src/main/scala/play/api/libs/json/JsValue.scala

```diff
@@ -134,15 +134,24 @@ case class JsObject(
   /**
    * The value of this JsObject as an immutable map.
    */
+  @deprecated("Use `valueMap` instead", "2.9.3")
   lazy val value: Map[String, JsValue] = underlying match {
     case m: immutable.Map[String, JsValue] => m
     case m                                 => m.toMap
   }
 
+  /**
+   * The value of this JsObject as an immutable map.
+   */
+  lazy val valueMap: immutable.Map[String, JsValue] = underlying match {
+    case m: immutable.Map[String, JsValue] => m
+    case m                                 => m.toMap
+  }
+
   /**
    * Return all fields as a set
    */
-  def fieldSet: Set[(String, JsValue)] = fields.toSet
+  def fieldSet: immutable.Set[(String, JsValue)] = fields.toSet
```

Comment:
If `Map` and `Set` there were the mutable one, then the same deprecation should be applied for `Set`